c2corg_api/views/document_associations.py

```diff
@@ -12,22 +13,23 @@
 from c2corg_api.views.document_schemas import waypoint_documents_config, \
     route_documents_config, user_profile_documents_config, \
     image_documents_config, article_documents_config, area_documents_config, \
-    outing_documents_config
+    outing_documents_config, book_documents_config
 from c2corg_api.views.validation import updatable_associations
 from sqlalchemy.sql.expression import or_, and_
 
 associations_to_include = {
     WAYPOINT_TYPE: {
         'waypoints', 'waypoint_children', 'routes', 'images', 'articles'},
     ROUTE_TYPE:
-        {'waypoints', 'routes', 'images', 'articles'},
+        {'waypoints', 'routes', 'images', 'articles', 'books'},
     OUTING_TYPE:
         {'waypoints', 'routes', 'images', 'users', 'articles'},
     IMAGE_TYPE:
         {'waypoints', 'routes', 'images', 'users', 'outings', 'articles',
-         'areas'},
+         'areas', 'books'},
     ARTICLE_TYPE:
-        {'waypoints', 'routes', 'images', 'users', 'outings', 'articles'},
+        {'waypoints', 'routes', 'images', 'users', 'outings', 'articles',
+         'books'},
     AREA_TYPE: {'images'},
     USERPROFILE_TYPE: {'images'}
```

Comment:
If you want that associations are included when requesting the details for a book, you will have to add an entry `BOOK_TYPE: {...}`.